db/db_impl/db_impl.cc

```diff
@@ -4092,7 +4094,8 @@ Status DBImpl::GetSuperSnapshots(
   for (auto& cf : column_families) {
     auto cfh = static_cast_with_check<ColumnFamilyHandleImpl>(cf);
     auto cfd = cfh->cfd();
-    auto sv = cfd->GetReferencedSuperVersion(this);
+    auto sv = cfd->GetSuperVersion();
+    sv->Ref();
```

Comment:
Not sure what the difference is here could you give me some context?

Reply:
They are equivalent, but `GetReferencedSuperVersion()` is meant to (and optimized to) be called outside of the lock. By adding a mutex to `GetSuperSnapshots()` (oops) it caused a deadlock :)

Reply:
It's using quite an interesting technique: https://github.com/rockset/rocksdb-cloud/blob/43810fe387e6746f3c96aea0cec51144652dda02/db/column_family.cc#L1260-L1277

Reply:
Will have to read more outside the function but definitely interesting. Not quite clear to me how thread local super versions get cleaned after a swap since local_sv becomes obsolete (dump thread does it? but then wouldn't that mess up the currently used sv for the thread?)